src/libraries/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs

```diff
@@ -183,6 +185,21 @@ public async Task ConnectAsync(Uri uri, CancellationToken cancellationToken, Cli
                     }
                 }
 
+                // Because deflate options are negotiated we need a new object
+                WebSocketDeflateOptions? negotiatedDeflateOptions = null;
+
+                if (options.DangerousDeflateOptions is not null && response.Headers.TryGetValues(HttpKnownHeaderNames.SecWebSocketExtensions, out IEnumerable<string>? extensions))
+                {
+                    foreach (ReadOnlySpan<char> extension in extensions)
+                    {
+                        if (extension.TrimStart().StartsWith(ClientWebSocketDeflateConstants.Extension))
+                        {
+                            negotiatedDeflateOptions = ParseDeflateOptions(extension, options.DangerousDeflateOptions);
```

Comment:
The issue here is that, because the method is async, we can't have ReadOnlySpan<char> variable. I can, however, move the entire thing into `ParseDeflateOptions`, making it return `bool` and have an `out` parameter for the parsed options.

Reply:
I see, in such case please ignore my comment and keep it as it is.